syntax = "proto3";

package ecommerce;
option java_package = "com.techstart.protobuf.ecommerce";
option java_multiple_files = true;

import "ecommerce/packet_type.proto";

enum ItemType{
  NONE =0;
  BIT_DATA=1;
  SHORT_DATA=2;
}

message ECommerceData{
  //  string item_name=1;
  int32 uniq_id=1;
  ecommerce.ItemType item_type=2;
  bool bit_value=3;
  int32 short_value=4;
}

message DataResponse{
  string msg_type=1;
  repeated ecommerce.ECommerceData ecommerce_data=2;
}

message DataRequest{
  ecommerce.PacketType packet_type=1;
}

service DataService{
  rpc getData(DataRequest) returns (stream DataResponse);
}

